src/native/corehost/fxr/command_line.cpp

```diff
@@ -164,6 +164,17 @@ namespace
                 trace::verbose(_X("Application '%s' is not a managed executable."), app_candidate.c_str());
                 if (!exec_mode)
                 {
+                    // Check if this is a non-managed file with directory separator that exists
+                    // This should show a specific error instead of routing to CLI
+                    bool has_dir_separator = app_candidate.find(DIR_SEPARATOR) != pal::string_t::npos;
+                    if (has_dir_separator)
+                    {
+                        if (pal::file_exists(app_candidate))
+                        {
+                            trace::error(_X("The application '%s' is not a managed .dll or .exe."), app_candidate.c_str());
```

Comment:
```suggestion
                            trace::error(_X("The application '%s' is not a managed .dll."), app_candidate.c_str());
```
I do not think managed .exes are a thing in modern .NET. None of the tooling will produce them. Should we drop `or .exe` here to avoid suggesting that managed .exes are a thing?

Alternatively, we can say `is not a .NET binary.". "binary" looks less Windows-specific, but also less descriptive. It won't give you a hint that you need to pass in .dll.